src/building/suggested.md

```diff
@@ -1,14 +1,39 @@
 # Suggested Workflows
 
-The full bootstrapping process takes quite a while. Here are three suggestions
+The full bootstrapping process takes quite a while. Here are five suggestions
 to make your life easier.
 
+## Configuring `rust-analyser` for `rustc`
+
+`rust-analyser` can help you check and format your code whenever you save
+a file. By default, `rust-analyser` runs the `cargo check` and `rustfmt`
+commands, but you can override these commands to use more adapted versions
+of these tools when hacking on `rustc`. For example, for Visual Studio Code,
+you can write:
+
+```JSON
+{
+    "rust-analyzer.checkOnSave.overrideCommand": [
+        "./x.py",
+        "check",
+        "--json-output"
+    ],
+    "rust-analyzer.rustfmt.overrideCommand": [
+      "./build/TARGET_TRIPLE/stage0/bin/rustfmt"
+    ],
+    "editor.formatOnSave": true
+}
+```
+
+in your `.vscode/settings.json` file. This will ask `rust-analyser` to use
+`x.py check` to check the sources, and the stage 0 rustfmt to format them.
```

Comment:
IIUC, you need to swap this configuration in and out to work in rustc and other Rust projects?  Or is this a project-specific config?

Reply:
VSCode allows for workspace-specific configuration, which is stored in the `.vscode` folder for your workspace. The sentence is probably unclear though, any suggestions?

Reply:
I will play with it in my copy and let you know if I have thoughts.  Initial thought is that .vscode dir needs to be in rustc gitignore or developer global ignore settings.  

Thanks @LeSeulArtichaut!

Reply:
and thanks for these instructions btw.  I have been looking for an approach to use vs code + rust analyzer with rustc.  👍

Reply:
Hehe, I only did the writeup for this. I didn't know any of this either before writing 😄
The worktrees stuff is going to be sooooo useful to me! I'm tired of being scared of building `rustc` because I want another build to stay there.

Reply:
I vaguely recall that git will default to ignoring files/dirs that start with `.`, so you might not need to add .vscode to gitignore...